### DISCUSSION TITLE
Get/calculate stator inductances

### DISCUSSION THREAD

**Discussion by carbon-drive** [Eomys/pyleecan#391] [Q&A] [answered]
Dear Pyleecan-Team,

first I want to thank you for your great packge! I am a mechanical engineer trying to build as light as possible motors. So far we where only able to reduce weight in substituting the steel shaft etc. with cfrp. Obviously if you eliminate all steel from your motor, you loose a lot of magnetic flux. So for the electromagnetic calculations on how much steel is needed I am so far totally dependent on the manufacturers of our stators and rotors. With pyleecan my hope is to accelerate and optimize this process by giving the manufactures an as good as possible design on which they can built. I played already a lot with different geometries and really like how easy it is to generate and calculate new geometries. 

I am struggling with a simple problem. I want to calculate the inductances. I thought I would find them in the "out.elec" object of the [Electric Tutorial](https://www.pyleecan.org/tuto_Elec.html). Or do I have to calculate them from Id and Ud?

Thanks again for your efforts!
Martin

**Comment by BonneelP** [comment, marked as answer]
Hello Martin,

Thank you for your interest for pyleecan :)

Great to know that you challenge the geometry modeller of pyleecan. If it is not confidential, can you share which kind of geometry experimentation you are working on ? I'm always curious to see how the different part of pyleecan can be combined to create new machines :) If you are looking for some strange shape you can play with uneven bore shapes, notches (on bore and/or yoke) and slot and hole from dxf. For optimisation, you could create a ParamExplorer with a user defined script to move point per point the geometry of your slot or hole or use the rotate/translate methods to reshape it. If need you can find some inspiration here: https://pyleecan.org/gallery.html. You can also create a DataKeeper with the mass of your machine (to have it accessible in plot_multi). 

Regarding the inductances, they are computed in the electrical module as part of the equivalent electrical circuit (EEC). So first you can select the model for computing them (for now we only have a model with FEMM with IndMagFEMM). The corresponding code is here: https://github.com/Eomys/pyleecan/tree/master/pyleecan/Methods/Simulation/IndMagFEMM. 
The inductances are part of the EEC and all the EEC parameters are gathered in EEC_PMSM.parameters (to be enforced or computed). EEC_PMSM.comp_parameters will fill the missing parameters in the dict and at the end of the simulation you should be able to access it with output.simu.elec.eec.parameters, keys as follow (from EEC_PMSM.comp_parameters):
![image](https://user-images.githubusercontent.com/20533034/119719651-54e3f080-be69-11eb-879e-c9650f41e941.png)

By taking a look on this part of the code I see that it can be improved: we shouldn't be editing a simulation property and the full dictionary should be stored in OutElec. I will update this part of the code once I will be sure that your questions is answered ;) Is it what you were looking for ? Do you need any other information ? 

Best regards,
Pierre

**Reply by Rutherfordio** [reply]
One question, when running a VarLoadCurrent simulation from a defined simulation that only has defined a magnetic simulation with MagFEMM, how is the inductance calculated then? And can we retrieve that inductance?
Edit: Never mind, figured it out.

**Comment by carbon-drive** [comment]
Hello Pierre,thank you for your quick response! 
My overall goal is quite simple. I am trying to find the motor with the least mass, that fullfills certain criteria for example:
- an S1 operating power of 25 kW  over a speed range of up to 20.000 rpm 
- with an initial torque of 60 Nm and a maximum current of 80 A
- with a maximum short term current of 140 A enough torque to accelerate in 0.5 sec to 20.000 rpm
On a more basic level I am also looking into geometries, that utilize our carbon fiber material as flux-barriers to create synRM motors, that are not possible with a pure steel design. Attached you find a picture of one quite ordinary looking motor. But for my calculations I use "Air" as rotor material and "M400-50A" as Magnet Material, thus creating a SynRM machine out of an SPMSM design.

![grafik](https://user-images.githubusercontent.com/17007547/119859713-ec068200-bf15-11eb-861f-35c31d8f1e95.png)
